### PR TITLE
fix(DB/SAI): Add Grizzly hills "log ride" transport

### PR DESCRIPTION
**Pull request by grimgravy**
## CHANGES PROPOSED:
-  In Grizzly Hills: Alliance and Horde transports added, going towards the two bases in Venture Bay
-  how it should work https://wow.gamepedia.com/Gordun

## ISSUES ADDRESSED:
- The transports don't work.

## TESTS PERFORMED:
- windows 7 64bits.
- works perfectly in the game.

## HOW TO TEST THE CHANGES:
 - .go creature 207212 
 - .go creature 98947

## Target branch(es):
- [x] grizzly_hills

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by BarbzYHOOL** [comment]
wow so cool I didn't even know this

The tree trunk is auto guided lol and goes very fast (I suppose it's the client who runs this?)

Tested on Horde side and works fine
Not tested on alliance side

**Comment by grimgravy** [comment]
taxinote.dbc, taxipath.dbc and taxipathnote.dbc. they are responsible for this.

**Comment by pklloveyou** [comment]
经过测试， 他们都工作得很 好 。A、H

**Comment by FrancescoBorzi** [comment]
good job @grimgravy !

thanks @pklloveyou for testing this

thanks @BarbzYHOOL for half-testing this xD

File: data/sql/updates/pending_db_world/rev_1584283998126323200.sql [hunk 1]
```diff
@@ -0,0 +1,24 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1584283998126323200');
+
+-- Grizzly Hills, Alliance Log Ride Start
+-- npc gordun
+DELETE FROM `gossip_menu_option` WHERE `MenuID`=21251 AND `OptionID`=1;
+INSERT INTO `gossip_menu_option` (`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES (21251, 1, 0, 'Yes, I am ready to travel to Venture Bay!', 26605, 1, 1, 0, 0, 0, 0, NULL, 0, 0);
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1584283998126323200.sql]:
```suggestion
INSERT INTO `gossip_menu_option` (`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES 
(21251, 1, 0, 'Yes, I am ready to travel to Venture Bay!', 26605, 1, 1, 0, 0, 0, 0, NULL, 0, 0);
```

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1584283998126323200.sql]: Just for consistency and easier to review the SQL this way imo.

Great job with this!

**Comment by Remberance** [comment]
win10 x60 test is ok!!!

File: data/sql/updates/pending_db_world/rev_1584283998126323200.sql [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 27425;
+
+DELETE FROM `smart_scripts` WHERE (`source_type` = 0 AND `entryorguid` = 27425);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(27425, 0, 0, 1, 62, 0, 100, 0, 9528, 0, 0, 0, 0, 72, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Darrok - On Gossip Option 0 Selected - Close Gossip'),
+(27425, 0, 1, 2, 61, 0, 100, 0, 0, 0, 0, 0, 0, 85, 48960, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Darrok - On Gossip Option 0 Selected - Invoker Cast \'Horde Log Ride 01 Begin\''),
+(27425, 0, 2, 0, 61, 0, 100, 0, 0, 0, 0, 0, 0, 85, 48961, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Darrok - On Gossip Option 0 Selected - Invoker Cast \'Log Ride Horde 00\'');
+
+-- quest Metzen the Reindeer
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1584283998126323200.sql]: @grimgravy why is this related with the PR?

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1584283998126323200.sql]: It actually makes no sense either because it is exactly like the DB is now, except with a different MenuID.
```sql
SELECT * FROM gossip_menu_option WHERE menuid IN (21251, 6763)
```

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1584283998126323200.sql]: Personally I think the Metzen changes should be in its own PR if these changes are necessary since it has no connection to the original intent of this pr? 🤷‍♂

Comment (grimgravy) [reply data/sql/updates/pending_db_world/rev_1584283998126323200.sql]: @BarbzYHOOL, had two types of creature using the same menuid (21251). Conflicts between them.

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1584283998126323200.sql]: The change is ok, but it does not have anything to do with the main issue of the PR, which is why I wondered if it could be in a separate PR ^^

Comment (FrancescoBorzi) [reply data/sql/updates/pending_db_world/rev_1584283998126323200.sql]: I agree with @Kitzunu also it is not good to push things to a PR that has already been tested, unless really important.

@grimgravy  can you please revert this PR to its original state? and push the other changes in another PR

**Comment by pklloveyou** [comment]
```
- INSERT INTO `conditions` VALUES (15, 21251, 0, 0, 0, 2, 0, 21211, 1, 0, 0, 0, 0, '', 'Metzen the Reindeer: Gossip option - player should have item Pouch of Reindeer Dust in inventory');
+INSERT INTO `conditions` VALUES (15, 6763, 0, 0, 0, 2, 0, 21211, 1, 0, 0, 0, 0, '', 'Metzen the Reindeer: Gossip option - player should have item Pouch of Reindeer Dust in inventory');
```

**Comment by Kitzunu** [comment]
@pklloveyou if you add \`\`\`sql it will get a sql tag for the code block
```sql
DELETE FROM `conditions` WHERE (`SourceTypeOrReferenceId` = 15 AND `SourceGroup` = 6763);

**Comment by grimgravy** [comment]
@pklloveyou thanks for your help.

File: data/sql/updates/pending_db_world/rev_1584283998126323200.sql (hunk 2 above)
Comment (BarbzYHOOL) [reply data/sql/updates/pending_db_world/rev_1584283998126323200.sql]: @grimgravy can you move all after ` -- the change was necessary because of the conflicts between MenuID existing in the DB.` to another PR?

Comment (grimgravy) [reply data/sql/updates/pending_db_world/rev_1584283998126323200.sql]: @FrancescoBorzi, has been settled

Comment (BarbzYHOOL) [reply data/sql/updates/pending_db_world/rev_1584283998126323200.sql]: nothing has been settled

Comment (grimgravy) [reply data/sql/updates/pending_db_world/rev_1584283998126323200.sql]: @FrancescoBorzi and @BarbzYHOOL if I do that, I'm gonna have to change here too.

Comment (BarbzYHOOL) [reply data/sql/updates/pending_db_world/rev_1584283998126323200.sql]: explain why clearly please

what was the conflict BEFORE this PR

Comment (FrancescoBorzi) [reply data/sql/updates/pending_db_world/rev_1584283998126323200.sql]: @grimgravy I can revert it for you

Comment (grimgravy) [reply data/sql/updates/pending_db_world/rev_1584283998126323200.sql]: alli

Comment (BarbzYHOOL) [reply data/sql/updates/pending_db_world/rev_1584283998126323200.sql]: if someone understands then do something plz (shin?)

**Comment by FrancescoBorzi** [comment]
I don't have access to modify this PR.

This is the original code that was tested and was ready to be merged:

```
INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1584283998126323200');

-- Grizzly Hills, Alliance Log Ride Start
-- npc gordun
DELETE FROM `gossip_menu_option` WHERE `MenuID`=21251 AND `OptionID`=1;
INSERT INTO `gossip_menu_option` (`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES (21251, 1, 0, 'Yes, I am ready to travel to Venture Bay!', 26605, 1, 1, 0, 0, 0, 0, NULL, 0, 0);

UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 27414;

DELETE FROM `smart_scripts` WHERE (`source_type` = 0 AND `entryorguid` = 27414);
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(27414, 0, 0, 1, 62, 0, 100, 0, 21251, 1, 0, 0, 0, 72, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Gordun - On Gossip Option 1 Selected - Close Gossip'),
(27414, 0, 1, 2, 61, 0, 100, 0, 0, 0, 0, 0, 0, 85, 48622, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Gordun - On Gossip Option 1 Selected - Invoker Cast \'Alliance Log Ride 01 Begin\''),
(27414, 0, 2, 0, 61, 0, 100, 0, 0, 0, 0, 0, 0, 85, 48621, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Gordun - On Gossip Option 1 Selected - Invoker Cast \'Log Ride Alliance 01\'');

-- Grizzly Hills, Horde Log Ride Start
-- npc darrok
UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 27425;

DELETE FROM `smart_scripts` WHERE (`source_type` = 0 AND `entryorguid` = 27425);
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(27425, 0, 0, 1, 62, 0, 100, 0, 9528, 0, 0, 0, 0, 72, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Darrok - On Gossip Option 0 Selected - Close Gossip'),
(27425, 0, 1, 2, 61, 0, 100, 0, 0, 0, 0, 0, 0, 85, 48960, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Darrok - On Gossip Option 0 Selected - Invoker Cast \'Horde Log Ride 01 Begin\''),
(27425, 0, 2, 0, 61, 0, 100, 0, 0, 0, 0, 0, 0, 85, 48961, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Darrok - On Gossip Option 0 Selected - Invoker Cast \'Log Ride Horde 00\'');
```

@grimgravy can you please just copy paste it into your sql file (remove everything else)

File: data/sql/updates/pending_db_world/rev_1584403279317487100.sql [hunk 3]
```diff
@@ -0,0 +1,26 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1584403279317487100');
+
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1584283998126323200');
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1584403279317487100.sql]: this line is duplicate, only one should be added

Comment (BarbzYHOOL) [reply data/sql/updates/pending_db_world/rev_1584403279317487100.sql]: i think grimgravy took the wrong one (should take the first one to match the filename)

**Comment by Kitzunu** [comment]
Thank you @grimgravy for the PR!

Thanks @pklloveyou for the test and @FrancescoBorzi & @BarbzYHOOL for the reviews ^^

**Comment by BarbzYHOOL** [comment]
:'(

i improved the title but you didn't use the new title when squashing :(

**Comment by grimgravy** [comment]
> : '(
> 
> Eu melhorei o título, mas você não usou o novo título ao esmagar :(

I'm sorry, I didn't pay attention that you had changed
